core/controllers/base_test.py

```diff
@@ -252,3 +255,25 @@ def test_logout_page(self):
         self.assertTrue(
             datetime.datetime.utcnow() > datetime.datetime.strptime(
                 expiry_date[1], '%a, %d %b %Y %H:%M:%S GMT',))
+
+
+class I18nDictsTest(test_utils.GenericTestBase):
+
+    def _extract_keys_from_json_file(self, filename):
+        return sorted(json.loads(utils.get_file_contents(
+            os.path.join(os.getcwd(), 'i18n', filename)
+        )).keys())
+
+    def test_i18n_keys(self):
+        """Tests that all JSON files in i18n.js have the same set of keys."""
+        master_key_list = self._extract_keys_from_json_file('en.json')
+        self.assertGreater(len(master_key_list), 0)
+
+        filenames = os.listdir(os.path.join(os.getcwd(), 'i18n'))
+        for filename in filenames:
+            if filename == 'en.json':
+                continue
+
+            key_list = self._extract_keys_from_json_file(filename)
+            # All other JSON files should have a subset of the keys in en.json.
+            self.assertEqual(len(set(key_list) - set(master_key_list)), 0)
```

Comment:
Of course you would and should remove unused keys from en.json. Unused keys are harmless in translation files if you use translatewiki.net and they will get removed when there are updates to the files.